src/lib/components/molecules/option-picker/index.jsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useCallback } from "preact/hooks"
+import { mergeStyles } from "$styles/helpers/mergeStyles"
+import defaultStyles from "./style.module.css"
+
+/** @typedef {  "vertical" | "horizontal" } OptionLayoutDirection */
+
+export function OptionPicker({
+  title,
+  options,
+  layoutDirection = "horizontal",
+  onSelect,
+  styles,
+}) {
+  styles = mergeStyles(defaultStyles, styles)
+
+  const [selectedIndex, setSelectedIndex] = useState(0)
+
+  const onOptionClick = useCallback(
+    (option, index) => {
+      if (index === selectedIndex) return
+
+      setSelectedIndex(index)
+      if (onSelect) onSelect(index, option)
+    },
+    [selectedIndex, onSelect],
+  )
+
+  return (
+    <div className={styles.optionPicker}>
+      {title && <span className={styles.title}>{title}</span>}
+      <div className={[styles.options, layoutDirection].join(" ")}>
```

Comment:
Suggestion: we could use the listbox role for this, and option roles for the children, to make this control more accessible to screen reader users. [We'd also need to manage a couple of attributes to do this _properly_](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role#required_javascript_features), it's up to you whether it's worth the time!